### ISSUE TITLE
scylla_setup failed to setup with error /etc/bashrc: line 99: TMOUT: readonly variable

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylla#8049]
During job: https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-oel81-test/3/execution/node/51/log/
Next scylla version was installed on Oracle Enterprise Linux 8.1 (ami-0e883e67243efea62 (eu-west-1)):
Scylla packages installed:
- scylla-node-exporter-4.5.dev-0.20210207.e1261d10f.x86_64
- scylla-tools-4.5.dev-0.20210207.e1261d10f.noarch
- scylla-conf-4.5.dev-0.20210207.e1261d10f.x86_64
- scylla-python3-4.5.dev-0.20210207.e1261d10f.x86_64
- scylla-jmx-4.5.dev-0.20210207.e1261d10f.noarch
- scylla-kernel-conf-4.5.dev-0.20210207.e1261d10f.x86_64
- scylla-tools-core-4.5.dev-0.20210207.e1261d10f.noarch
- scylla-4.5.dev-0.20210207.e1261d10f.x86_64
- scylla-server-4.5.dev-0.20210207.e1261d10f.x86_64
from repository: https://s3.amazonaws.com/downloads.scylladb.com/unstable/scylla/master/rpm/centos/2021-02-07T14:29:36Z/scylla.repo

But scylla_setup command failed with:
```
Command: 'sudo /usr/lib/scylla/scylla_setup --nic ens3 --disks /dev/nvme0n1 --setup-nic-and-disks  --swap-directory / '
 
 Exit code: 1
 
 Stdout:
 
 [1;31mScylla does not work with current umask setting (0077),
 please restore umask to the default value (0022).[0m
 
 Stderr:
 
 /etc/bashrc: line 99: TMOUT: readonly variable
 /etc/bashrc: line 100: TMOUT: readonly variable
 ```

**Comment by syuu1228** [comment]
I couldn't able to run ami-0e883e67243efea62 since I do not find login user information of the AMI, but I think the environment has non-standard umask configuration, as scylla_setup says.
Only solution to fix it is let user to restore default umask setting manually, since we don't want to change it automatically.

Related: https://github.com/scylladb/scylla/commit/60d8a99f056465cc925f3fa1a21552f95cf64162
Related: #4157

I did not reproduce the issue on Oracle Linux 8.3 Vagrant image, BTW.

**Comment by syuu1228** [comment]
Note: Very likely " TMOUT: readonly variable" can ignore, why exiting with non-zero is umask setting.

**Comment by syuu1228** [comment]
Sorry, login failure on the AMI was probably my local environment issue, I able to login after I switched another PC.

BTW, I found why the error happen: the AMI is modifying umask parameter from default setting, as I described at https://github.com/scylladb/scylla/issues/8049#issuecomment-777153378.
Both RHEL8 and OEL8 should have default umask = 0022, like this (it's on OEL8 on Vagrant):
```
[vagrant@localhost ~]$ grep umask /etc/bashrc /etc/profile
/etc/bashrc:    # By default, we want umask to get set. This sets it for non-login shell.
/etc/bashrc:       umask 002
/etc/bashrc:       umask 022
/etc/profile:# By default, we want umask to get set. This sets it for login shell
/etc/profile:    umask 002
/etc/profile:    umask 022
```

However, on the AMI it has different value:
```
[ec2-user@ip-10-0-0-83 ~]$ grep umask /etc/bashrc /etc/profile
/etc/bashrc:    # By default, we want umask to get set. This sets it for non-login shell.
/etc/bashrc:       umask 077
/etc/bashrc:       umask 077
/etc/profile:# By default, we want umask to get set. This sets it for login shell
/etc/profile:    umask 077
/etc/profile:    umask 077
```
To run Scylla correctly, last digit of umask should be '2', so scylla_setup is failing on this environment.

I compared these files with rpm provided version, it is modified when AMI created,
different with default:
```
[ec2-user@ip-10-0-0-83 ~]$ mkdir tmp
[ec2-user@ip-10-0-0-83 ~]$ cd tmp
[ec2-user@ip-10-0-0-83 tmp]$ yumdownloader setup-2.12.2-2.el8.noarch
Last metadata expiration check: 0:09:33 ago on Sat Feb 13 09:58:36 2021.
setup-2.12.2-2.el8.noarch.rpm                   2.7 MB/s | 180 kB     00:00    
[ec2-user@ip-10-0-0-83 tmp]$ rpm2cpio setup-2.12.2-2.el8.noarch.rpm |cpio -idm
1424 blocks
[ec2-user@ip-10-0-0-83 tmp]$ diff -u ./etc/bashrc /etc/bashrc 
--- ./etc/bashrc	2018-09-10 11:51:03.000000000 +0000
+++ /etc/bashrc	2020-03-18 19:05:17.092312772 +0000
@@ -72,9 +72,9 @@
     # You could check uidgid reservation validity in
     # /usr/share/doc/setup-*/uidgid file
     if [ $UID -gt 199 ] && [ "`id -gn`" = "`id -un`" ]; then
-       umask 002
+       umask 077
     else
-       umask 022
+       umask 077
     fi
 
     SHELL=/bin/bash
@@ -96,3 +96,5 @@
 
 fi
 # vim:ts=4:sw=4
+readonly TMOUT=900 ; export TMOUT
+TMOUT=600
[ec2-user@ip-10-0-0-83 tmp]$ diff -u ./etc/profile /etc/profile
--- ./etc/profile	2018-09-10 11:51:03.000000000 +0000
+++ /etc/profile	2020-03-18 19:05:17.092312772 +0000
@@ -57,9 +57,9 @@
 # You could check uidgid reservation validity in
 # /usr/share/doc/setup-*/uidgid file
 if [ $UID -gt 199 ] && [ "`id -gn`" = "`id -un`" ]; then
-    umask 002
+    umask 077
 else
-    umask 022
+    umask 077
 fi
 
 for i in /etc/profile.d/*.sh /etc/profile.d/sh.local ; do
@@ -83,3 +83,5 @@
                 . /etc/bashrc
        fi
 fi
+readonly TMOUT=900 ; export TMOUT
+TMOUT=600
```

Also this is the reason why we see "TMOUT: readonly variable" error on stderr.

If we want keep using this AMI for artifact tests, I think we need to restore default configuration by following commands before running the test:
```
sudo mv /etc/bashrc /etc/bashrc.modified
sudo mv /etc/profile /etc/profile.modified
sudo yum reinstall -y setup
```

**Comment by syuu1228** [comment]
Note: As I already mentioned at https://github.com/scylladb/scylla/issues/8049#issuecomment-777153378, we should not auto-configure system umask since it's unsafe.
We may able to relax umask verification code to just a warning, not error exit, but then user may hit #4157.
To avoid both issue, I suppose we need to make all files owned by scylla, perhaps more script run in scylla (or chown file after running the script).

**Comment by syuu1228** [comment]
Or maybe it can resolve by chmod a+r for every files created by root.

**Comment by slivne** [comment]
@syuu1228 is this fixed ?

**Comment by syuu1228** [comment]
@slivne Not fixed, but we should do some fixup in scylla-cluster-tests, not in scylla main repo, at least on current implementation.
Because it's caused by unoffical OEL8 AMI (not published from Oracle and it is customized).

Alternatively we may able to redesign our scripts to run in such umask condition.

**Comment by syuu1228** [comment]
@avikivity what do you think about this? Should we able to run scylla scripts in umask=077?

**Comment by avikivity** [comment]
I suppose if we are running as root we can chgrp io_properties.conf to the scyllaadm group. Then scylla will be able to read it.

Another option is to just chmod it to be world-readable. It is not sensitive after all. So maybe we can change umask, if running as root, to 022.

 - non-root installs run under the original user and don't care about umask
 - root install works

**Comment by roydahan** [comment]
We need to backport this fix to 4.4

**Comment by avikivity** [comment]
It's already in 4.4 (f96ea8e01182865e05bf189bfb0c2f4e287d4320).

**Comment by yarongilor** [comment]
the issue is still reproduced in both master and 4.4 using:
http://downloads.scylladb.com/unstable/scylla/branch-4.4/rpm/centos/2021-03-04T13:53:40Z
and:
http://downloads.scylladb.com/unstable/scylla/master/rpm/centos/2021-03-07T21:45:44Z
@syuu1228 , can you please advise?

**Comment by roydahan** [comment]
@yarongilor i think it turned out not to be this issue but #8238

**Comment by roydahan** [comment]
Shall we backport the fix for that to 2021.1?

**Comment by avikivity** [comment]
Fully backported, removing backported candidate label.
